src/librbd/internal.cc

```diff
@@ -1870,6 +1870,9 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
       return r;
     }
 
+    // TODO: filter out internal metadata here?
```

Comment:
I'd vote to optionally hide keys prefixed w/ `.` via the CLI and let the API handle everything.